src/main/java/sirius/biz/scripting/mongo/MongoCustomEventDispatcherRepository.java

```diff
@@ -0,0 +1,107 @@
+import java.util.Optional;
+
+/**
+ * Stores and manages {@link CustomEventDispatcher custom event dispatchers} in a MongoDB.
+ */
+@Register(framework = MongoCustomEventDispatcherRepository.FRAMEWORK_SCRIPTING_MONGO)
+public class MongoCustomEventDispatcherRepository implements CustomEventDispatcherRepository {
+
+    protected static final String FRAMEWORK_SCRIPTING_MONGO = "biz.scripting-mongo";
+
+    /**
+     * Contains the name of the variable which holds the {@link CustomEventRegistry} in a script.
+     */
+    public static final String SCRIPT_PARAMETER_REGISTRY = "registry";
+
+    @Part
+    private Mango mango;
+
+    @Override
+    public List<String> fetchAvailableDispatchers(@Nonnull String tenantId) {
+        return mango.select(MongoCustomScript.class)
+                    .eq(MongoCustomScript.TENANT, tenantId)
+                    .orderAsc(MongoCustomScript.CODE)
+                    .queryList()
+                    .stream()
+                    .map(MongoCustomScript::getCode)
+                    .toList();
+    }
+
+    @Override
+    public Optional<CustomEventDispatcher> fetchDispatcher(@Nonnull String tenantId, @Nullable String name) {
+        if (Strings.isEmpty(name)) {
+            List<MongoCustomScript> mongoCustomScripts =
+                    mango.select(MongoCustomScript.class).eq(MongoCustomScript.TENANT, tenantId).limit(2).queryList();
+            if (mongoCustomScripts.size() == 1) {
+                return compileAndLoad(mongoCustomScripts.getFirst());
+            } else {
+                return Optional.empty();
+            }
+        } else {
+            return mango.select(MongoCustomScript.class)
+                        .eq(MongoCustomScript.TENANT, tenantId)
+                        .eq(MongoCustomScript.CODE, name)
+                        .first()
+                        .flatMap(this::compileAndLoad);
+        }
+    }
+
+    private Optional<CustomEventDispatcher> compileAndLoad(MongoCustomScript script) {
+        try {
+            if (Strings.isEmpty(script.getScript())) {
+                return Optional.empty();
+            }
+
+            CompilationContext compilationContext =
+                    new CompilationContext(SourceCodeInfo.forInlineCode(script.getScript(), SandboxMode.WARN_ONLY));
+
+            compilationContext.getVariableScoper()
+                              .defineVariable(Position.UNKNOWN, SCRIPT_PARAMETER_REGISTRY, CustomEventRegistry.class);
+            NoodleCompiler compiler = new NoodleCompiler(compilationContext);
+            Callable compiledScript = compiler.compileScript();
+
+            ScriptBasedCustomEventDispatcher dispatcher = new ScriptBasedCustomEventDispatcher();
+            SimpleEnvironment environment = new SimpleEnvironment();
+            environment.writeVariable(0, dispatcher);
+            compiledScript.call(environment);
+
+            return Optional.of(dispatcher);
+        } catch (ScriptingException | HandledException e) {
```

Comment:
**Optional:** Rename `e` to `exception` :)

Reply:
done